### DISCUSSION TITLE
Package Deprecation

### DISCUSSION THREAD

**Discussion by rawilk** [rawilk/laravel-app-key-rotator#24] [Announcements]
With the upcoming Laravel 11.x release, it looks like the problem this package was created to solve will be handled by the framework now with Laravel's new built-in ability to set new app keys, and set previous app keys to handle decrypting data that isn't encrypted yet with the new key.

With that ability written in the framework itself, this package isn't really necessary anymore, so I probably won't be maintaining it any further and I probably won't even add Laravel 11.x compatibility either.

**Comment by nunodonato** [comment]
@rawilk is that as simple as that? 
I'm using your package and we automated(scheduled) the app key rotation. It doesn't seem that laravel 11 allows that. It simply makes things a bit easier by supporting an older app key, but there's no built-in feature to make the rotation.
I'd very much prefer that this package is updated to support laravel 11, but I also understand if you prefer to discontinue it. In that case I'll do a fork to maintain the support for L11

**Comment by rawilk** [comment]
There's still a little work you need to do when you rotate keys in L11, however I don't think it necessarily warrants a package for it anymore. Laravel will handle decrypting data as long as you have your previous keys defined in `APP_PREVIOUS_KEYS`; you'll just need to re-encrypt any models that have encrypted fields if you want to remove that previous app key at some point.

Personally, I'm just going to write a simple scheduled command that does that, however in the future I may revisit this package to provide a command like that for you. For now though, I don't really have any interest in maintaining for Laravel 11 yet, so it's probably best to just fork your own.
